src/ce/crt0.src

```diff
@@ -143,21 +140,65 @@ assert mpTmr2Counter shr 8 = mpTmrCtrl shr 8
 	ld	l, mpTmrCtrl and $FF
 	set	bTmr2Enable, (hl)
 
+	section	.text
+	private	___rerun_handler
+	private	___rerun_handler.callback
+	private	___rerun_handler.data
+	private	___rerun_handler.retval
+___rerun_handler:
+	ld	hl,(_exit.sp)
+	ld	de,0
+.data_size := $ - 3
+	add	hl,de
+	ld	(_exit.sp),hl		; move exit.sp to real ret
+	ld	hl,0
+.retval := $ - 3
+	ex	(sp),hl
+	ld	de,0
+.data := $ - 3
+	push	de,hl
+	jp	0
+.callback := $ - 3
+
+	section	.init.reruncheck
+	public	___prgm_run_check
+___prgm_run_check:
+	ld	de,___prgm_caller
+	call	002032Ch		; ti.MovFROP1
+	pop	hl
+	ld	de,-1
+	or	a,a
+	sbc	hl,de
+	add	hl,de
+	jr	nz,.no_return
+	ld	de,___rerun_handler
+	ld	(__start._main),de	; doesn't work for custom start?
+	pop	de
+	ld	(___rerun_handler.retval),de
+	pop	de
+	ld	(___rerun_handler.callback),de
+	pop	de
+	ld	(___rerun_handler.data_size),de
+	or	a,a
+	sbc	hl,hl
+	add	hl,sp			; user callback data
+	ld	(___rerun_handler.data),hl
+	jq	__start
+.no_return:
+	push	hl
```

Comment:
`__start._main` should probably be a weak symbol accessible from C so that it can define its own to call indirectly and not have to have the correct declaration of main visible while also supporting rerun.

Also, actually working optimization:
```suggestion
	ld	de,1
	add	hl,de
	dec	hl
	jq	nc,.no_rerun
	ld	hl,___rerun_handler
	ld	(__start._main),hl	; doesn't work for custom start?
	pop	hl
	ld	(___rerun_handler.retval),hl
	pop	hl
	ld	(___rerun_handler.callback),hl
	pop	hl
	ld	(___rerun_handler.data_size),hl
	ex	de,hl
	add	hl,sp			; user callback data
	dec	hl
	ld	(___rerun_handler.data),hl
virtual
	ld	a,0
 load .ld_a: byte from $$
end virtual
	db	.ld_a
.no_rerun:
	push	hl
```